charms/tensorboards-web-app/src/charm.py

```diff
@@ -1,158 +1,252 @@
+            f" --bind 0.0.0.0:{self._http_port}"
+            " --access-logfile"
+            " - entrypoint:app"
+        )
 
-        self.model.unit.status = MaintenanceStatus("Setting pod spec")
-
-        self.model.pod.set_spec(
-            {
-                "version": 3,
-                "serviceAccount": {
-                    "roles": [
-                        {
-                            "global": True,
-                            "rules": [
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["namespaces"],
-                                    "verbs": ["get", "list"],
-                                },
-                                {
-                                    "apiGroups": ["authorization.k8s.io"],
-                                    "resources": ["subjectaccessreviews"],
-                                    "verbs": ["create"],
-                                },
-                                {
-                                    "apiGroups": ["tensorboard.kubeflow.org"],
-                                    "resources": [
-                                        "tensorboards",
-                                        "tensorboards/finalizers",
-                                    ],
-                                    "verbs": ["get", "list", "create", "delete"],
-                                },
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["persistentvolumeclaims"],
-                                    "verbs": ["create", "delete", "get", "list"],
-                                },
-                                {
-                                    "apiGroups": ["storage.k8s.io"],
-                                    "resources": ["storageclasses"],
-                                    "verbs": ["get", "list", "watch"],
-                                },
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": ["poddefaults"],
-                                    "verbs": ["get", "list", "watch"],
-                                },
-                            ],
-                        }
-                    ]
+        layer_config = {
+            "summary": "jupyter-web-app layer",
+            "description": "Pebble config layer for jupyter-web-app",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Entrypoint of jupyter-web-app image",
+                    "command": exec_command,
+                    "startup": "enabled",
+                    "environment": self._env_vars,
+                    "on-check-failure": {"up": "restart"},
+                }
+            },
+            "checks": {
+                "up": {
```

Comment:
I believe a check named `up` could be easily confused with the default container agent ones (e.g. `up`, `alive`, `ready`). Would a more explicit name (derived from your container name) make more sense to you?
```suggestion
                    "on-check-failure": {"tensorboards-web-app-up": "restart"},
                }
            },
            "checks": {
                "tensorboards-web-app-up": {
```

Reply:
Sure, I 'd go with the proposed change